lib/rules/function-url-scheme-blacklist/__tests__/index.js

```diff
@@ -0,0 +1,218 @@
+  }, {
+    code: "a { background: some-url(); }",
+    description: "ignore contain url function",
+  }, {
+    code: "a { background: url($image); }",
+    description: "ignore variable",
+  }, {
+    code: "a { background: url(@image); }",
+    description: "ignore variable",
+  }, {
+    code: "a { background: url(http://#{$host}/path); }",
+    description: "ignore interpolation",
+  }, {
+    code: "a { background: url('http://@{host}/path'); }",
+    description: "ignore interpolation",
+  }, {
+    code: "a { background: url(http://$(host)/path); }",
+    description: "ignore interpolation",
+  }, {
+    code: "a { background: url(var(--image)); }",
+    description: "ignore variable",
+  }, {
+    code: "a { background: url(example.com); }",
+    description: "schemeless url",
+  }, {
+    code: "a { background: url(example.com:3000); }",
+    description: "schemeless url and port",
+  }, {
+    code: "a { background: url(http://example.com:3000); }",
+    description: "url with scheme and port",
+  } ],
+
+  reject: [ {
+    code: "a { background-image: url('data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs='); }",
+    message: messages.rejected("data"),
+    line: 1,
+    column: 27,
+  }, {
+    code: "a { background: url(https://www.example.com/file.jpg); }",
+    message: messages.rejected("https"),
+    line: 1,
+    column: 21,
+  }, {
+    code: "a { background: url('https://www.example.com/file.jpg'); }",
+    message: messages.rejected("https"),
+    line: 1,
+    column: 21,
+  }, {
+    code: "a { background: url(\"https://www.example.com/file.jpg\"); }",
+    message: messages.rejected("https"),
+    line: 1,
+    column: 21,
+  }, {
+    code: "a { background: url('https://example.com:3000'); }",
+    message: messages.rejected("https"),
+    line: 1,
+    column: 21,
+  }, {
+    code: "@font-face { font-family: 'foo'; src: url('https://www.example.com/file.jpg'); }",
+    message: messages.rejected("https"),
+    line: 1,
+    column: 43,
+  }, {
+    code: "a { background: no-repeat center/80% url('https://www.example.com/file.jpg'); }",
+    message: messages.rejected("https"),
+    line: 1,
+    column: 42,
+  } ],
+})
+
+testRule(rule, {
+  ruleName,
+  config: [[]],
+
+  accept: [ {
+    code: "a { background: url('/path/to/file.jpg'); }",
+  }, {
+    code: "a { background: url(//www.example.com/file.jpg); }",
+  }, {
+    code: "a { background: url(\"//www.example.com/file.jpg\"); }",
+  }, {
+    code: "a { background: url(example.com:3000); }",
+  } ],
+
+  reject: [ {
+    code: "a { background: url('https://www.example.com/file.jpg'); }",
+    message: messages.rejected("https"),
+    line: 1,
+    column: 21,
+  }, {
+    code: "a { background-image: url('data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs='); }",
+    message: messages.rejected("data"),
+    line: 1,
+    column: 27,
+  } ],
+})
+
+testRule(rule, {
+  ruleName,
+  config: [""],
```

Comment:
@hudochenkov can your help me with this, I would like the current time to concentrate on the new rules (https://github.com/stylelint/stylelint/issues/2528) and I'm bad at documentation :disappointed: